backend/blueprints/spa_api/service_layers/replay/tag.py

```diff
@@ -128,33 +132,30 @@ def get_encoded_private_key(name: str, session=None) -> str:
     @staticmethod
     def encode_tag(tag_id: int, private_id: str) -> str:
         merged = str(tag_id + 1000) + ":" + private_id
-        return base64.b85encode(merged.encode(encoding="utf-8")).decode('utf-8')
+        return quote(base64.b85encode(merged.encode(encoding="utf-8")).decode('utf-8'))
 
     @staticmethod
     def decode_tag(encoded_key: str):
-        decoded_key_bytes = base64.b85decode(encoded_key)
-        decoded_key = decoded_key_bytes.decode(encoding="utf-8")
+        decoded_key_bytes = base64.b85decode(unquote(encoded_key).encode('utf-8'))
+
+        try:
+            decoded_key = decoded_key_bytes.decode(encoding="utf-8")
+        except UnicodeDecodeError as e:
+            raise TagKeyError(encoded_key, e)
+
         first_index = decoded_key.find(':')
         tag_id = int(decoded_key[0: first_index]) - 1000
         decoded_private_id = decoded_key[first_index + 1:]
         return tag_id, decoded_private_id
 
+
 @with_session
-def apply_tags_to_game(query_params: Dict[str, any]=None, game_id=None, session=None):
+def apply_tags_to_game(query_params: Dict[str, any] = None, game_id=None, session=None):
     if query_params is None:
         return None
-    if 'tags' not in query_params and 'private_tag_keys' not in query_params:
+    if 'private_tag_keys' not in query_params:
```

Comment:
instead of returning None, maybe indicate that the request was malformed? Or if `query_params` is already verified to not be `None` and `private_tag_keys` is not optional, remove these `if`s?